app/src/main/java/org/wikipedia/history/HistoryFragment.java

```diff
@@ -413,6 +375,78 @@ void bindItem(@NonNull String date) {
         }
     }
 
+    private interface SearchHeaderCallback {
+        void onSearchCardClicked();
+
+        void onVoiceSearchClicked();
+
+        void onFilterHistoryClicked();
+
+        void onClearHistoryClicked();
+    }
+
+    private class SearchHeaderItemCallback implements SearchHeaderCallback {
+        @Override
+        public void onSearchCardClicked() {
+            ((MainFragment) getParentFragment()).openSearchActivity(Constants.InvokeSource.NAV_MENU, null);
+        }
+
+        @Override
+        public void onVoiceSearchClicked() {
+            ((MainFragment) getParentFragment()).onFeedVoiceSearchRequested();
+        }
+
+        @Override
+        public void onFilterHistoryClicked() {
+            if (actionMode == null) {
+                actionMode = ((AppCompatActivity) requireActivity())
+                        .startSupportActionMode(searchActionModeCallback);
+            }
+        }
+
+        @Override
+        public void onClearHistoryClicked() {
+            if (selectedEntries.size() == 0) {
+                new AlertDialog.Builder(requireContext())
+                        .setTitle(R.string.dialog_title_clear_history)
+                        .setMessage(R.string.dialog_message_clear_history)
+                        .setPositiveButton(R.string.dialog_message_clear_history_yes, (dialog, which) -> onClearHistoryClick())
+                        .setNegativeButton(R.string.dialog_message_clear_history_no, null).create().show();
+            } else {
+                deleteSelectedPages();
+            }
+        }
+    }
+
+    private class SearchCardViewHolder extends DefaultViewHolder<View> {
+        WikiCardView searchWikiCardView;
+        View searchCardView;
+        AppCompatImageView voiceSearchButton;
+        ImageView historyFilterButton;
+        ImageView clearHistoryButton;
+        SearchHeaderCallback searchHeaderCallback;
```

Comment:
Some varibles can be local varibles, and some varibles can add `private` on them.